# Generated by Django 2.2 on 2019-04-27 03:04

from django.db import migrations


def forwards_func(apps, schema_editor):
    """
    :param apps:
    :param schema_editor:
    :return:
    """
    department = apps.get_model("rbac", "Department")
    menu = apps.get_model("rbac", "Menu")

    db_alias = schema_editor.connection.alias
    try:
        department.objects.using(db_alias).bulk_create([
            department(id=1, title="运维部"),
            department(id=2, title="行政部"),
        ])
        menu.objects.using(db_alias).bulk_create([
            menu(id=1, name="用户与角色", types="system"),
            menu(id=2, name="菜单管理", types="system"),
            menu(id=3, name="权限和权限组", types="system"),
            menu(id=4, name="当前用户", types="current"),
        ])
    except Exception as error:
        print(error)


def reverse_func(apps, schema_editor):
    Department = apps.get_model("rbac", "Department")
    db_alias = schema_editor.connection.alias
    Department.objects.using(db_alias).filter(id=1).filter(name="运维部").delete()
    Department.objects.using(db_alias).filter(id=2).filter(name="行政部").delete()


class Migration(migrations.Migration):
    dependencies = [
        ('rbac', '0001_initial'),
    ]

    operations = [
        migrations.RunPython(forwards_func, reverse_func)
    ]
